src/collider/GeomTree.cpp

```diff
@@ -132,18 +112,40 @@ GeomTree::GeomTree(const int numVerts, const int numTris, const std::vector<vect
 		m_edges[pos].len = len;
 		m_edges[pos].dir = dir;
 
-		edgeIdxs[pos] = pos;
-		m_aabbs[pos].min = m_aabbs[pos].max = vector3d(v1);
-		m_aabbs[pos].Update(vector3d(v2));
+		// Build list of AABBs for edge BVH tree
+		AABBd edgeAabb = AABBd::Invalid();
+		edgeAabb.Update(vector3d(v1));
+		edgeAabb.Update(vector3d(v2));
+		m_edgeAABBs[pos] = edgeAabb;
 	}
 
-	//t = SDL_GetTicks();
-	m_edgeTree.reset(new BVHTree(m_numEdges, edgeIdxs, &m_aabbs[0]));
-	delete[] edgeIdxs;
-	//Output("Edge tree of %d edges build in %dms\n", m_numEdges, SDL_GetTicks() - t);
+	// Compute the bounds for edge/triangle BVH tree
+	AABBd bounds = AABBd { m_aabb.min, m_aabb.max};
 
-	timer.Stop();
-	//Output(" - - GeomTree::GeomTree took: %lf milliseconds\n", timer.millicycles());
+	{
+		PROFILE_SCOPED_DESC("GeomTree::CreateEdgeTree");
+		m_edgeTree.reset(new BinnedAreaBVHTree());
+		m_edgeTree->Build(bounds, m_edgeAABBs.get(), m_numEdges);
+	}
+
+	{
+		PROFILE_SCOPED_DESC("GeomTree::CreateTriTree");
+
+		m_triAABBs.reset(new AABBd[m_numTris]);
+		for (size_t i = 0; i < m_numTris; i++) {
```

Comment:
Warning about signs. (and in one more place)

Reply:
Existing code / behavior. Not going to address it here (see comment further down).